x-pack/filebeat/input/httpjson/input_manager.go

```diff
@@ -21,21 +23,30 @@ import (
 type inputManager struct {
 	stateless *stateless.InputManager
 	cursor    *cursor.InputManager
+
+	v2inputManager v2.InputManager
 }
 
-var _ v2.InputManager = inputManager{}
+var _ inputv2.InputManager = inputManager{}
 
 // Init initializes both wrapped input managers.
-func (m inputManager) Init(grp unison.Group, mode v2.Mode) error {
+func (m inputManager) Init(grp unison.Group, mode inputv2.Mode) error {
 	return multierr.Append(
-		m.stateless.Init(grp, mode),
-		m.cursor.Init(grp, mode),
+		multierr.Append(
+			m.stateless.Init(grp, mode),
+			m.cursor.Init(grp, mode),
+		),
+		m.v2inputManager.Init(grp, mode),
 	)
 }
 
 // Create creates a cursor input manager if the config has a date cursor set up,
 // otherwise it creates a stateless input manager.
-func (m inputManager) Create(cfg *common.Config) (v2.Input, error) {
+func (m inputManager) Create(cfg *common.Config) (inputv2.Input, error) {
+	if b, _ := cfg.Bool("is_v2", -1); b {
```

Comment:
Did you consider using string or int field for this? Like `config_version: 2`. Reminds me of https://www.vagrantup.com/docs/vagrantfile/version.html.

Reply:
That makes a lot of sense 👍